checks.d/dns_check.py

```diff
@@ -20,66 +20,96 @@
+            raise BadConfException("You must specify a hostname")
 
-    def query_dns(self, instance, timeout):
-        """Perform the DNS query, and report its duration as a gauge"""
-        start_time = end_time = 0.0
         resolver = dns.resolver.Resolver()
-        resolver.lifetime = timeout
 
-        hostname = instance.get('hostname')
-        tags = self._get_tags(instance)
-
-        # if a specific DNS server was defined use it, else use the system default
+        # If a specific DNS server was defined use it, else use the system default
         nameserver = instance.get('nameserver')
         if nameserver is not None:
             resolver.nameservers = [nameserver]
 
+        timeout = float(instance.get('timeout', self.default_timeout))
+        resolver.lifetime = timeout
         record_type = instance.get('record_type', 'A')
 
-        status = AgentCheck.CRITICAL
+        return hostname, timeout, nameserver, record_type, resolver
+
+    def _check(self, instance):
+        hostname, timeout, nameserver, record_type, resolver = self._load_conf(instance)
+
+        # Perform the DNS query, and report its duration as a gauge
+        start_time = end_time = 0.0
         start_time = time.time()
+
         try:
-            self.log.debug('Querying "%s" record for hostname "%s"...' % (record_type, hostname))
+            self.log.debug('Querying "{0}" record for hostname "{1}"...'.format(record_type, hostname))
             answer = resolver.query(hostname, rdtype=record_type)
             assert(answer.rrset.items[0].to_text())
             end_time = time.time()
+
         except dns.exception.Timeout:
-            self.log.error('DNS resolution of %s timed out' % hostname)
-            self.service_check(self.SERVICE_CHECK_NAME, status, tags=self._get_tags(instance))
-            raise
+            self.log.error('DNS resolution of {0} timed out'.format(hostname))
+            return Status.CRITICAL, 'DNS resolution of {0} timed out'.format(hostname)
+
         except Exception:
-            self.log.exception('DNS resolution of %s has failed.' % hostname)
-            self.service_check(self.SERVICE_CHECK_NAME, status, tags=self._get_tags(instance))
-            raise
+            self.log.exception('DNS resolution of {0} has failed.'.format(hostname))
+            return Status.CRITICAL, 'DNS resolution of {0} has failed'.format(hostname)
+
         else:
+            tags = self._get_tags(instance)
             if end_time - start_time > 0:
-                self.gauge('dns.response_time', end_time - start_time, tags=tags)
-                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=self._get_tags(instance))
+                self.gauge('dns.response_time', end_time - start_time, tags)
+            self.log.debug('Resolved hostname: {0}'.format(hostname))
+            return Status.UP, 'UP'
 
     def _get_tags(self, instance):
+        instance_name = instance.get('name')
```

Comment:
I think people upgrading from a previous version won't have a `name` so we should default to the `hostname`